docs/components/Layout/Layout.jsx

```diff
@@ -7,11 +7,34 @@ import Header from '../Header';
 import SideMenu from '../SideMenu';
 import lightTheme from '../../themes/light';
 import darkTheme from '../../themes/dark';
+import { getAutoTheme } from '../../utils';
 import { theme } from '../../../constants';
 import { MainContainer, AppContainer, MdxContainer, LinkText } from './styles';
 
 const Layout = ({ children }) => {
   const [currentTheme, setTheme] = useState(theme.dark);
+  const [autoTheme, setAutoTheme] = useState(false);
+
+  const saveTheme = (type) => {
+    setTheme(type);
+    sessionStorage.setItem('theme', type);
+  };
+
+  const checkTheme = (type) => {
+    if (type === theme.auto) {
+      setAutoTheme(true);
+      saveTheme(getAutoTheme());
+    } else {
+      setAutoTheme(false);
+      saveTheme(type);
+    }
+  };
+
+  useEffect(() => {
+    if (autoTheme) {
+      saveTheme(getAutoTheme());
+    }
+  }, [children]);
```

Comment:
I think there should be another dependency. Maybe `autoTheme`, but not `children` (it's strange).

Reply:
I created this function to check which theme should be used.
For example, user open the app at 20:59 (light theme still on) and we need to update the theme to a dark after 21:00. I made it so that if the user switches to another page (children at Layout component changes), theme will be "calculated" again and changed to dark.

Reply:
Then I think you can leave it without deps at all and it will be called every time.
```javascript
useEffect(() => { // your code })
```

Reply:
I'm not sure it will work ^

Reply:
Yes, I can delete deps array. Without deps, effects run after every render — and setting the state will trigger the effects again. If this is normal, then I will do so.